### PR TITLE
Adding slack links to community and discussion pages

### PR DESCRIPTION
**Pull request by chatman**
There are more options for support these days than the community page mentions. Lets add as much info as possible.

### CODE REVIEW

**Comment by chatman** [comment]
@epugh FYI, contains a link to the OSC slack.

File: content/pages/core/discussion.md [hunk 1]
```diff
@@ -86,7 +86,9 @@ but developers should be careful to transfer all the official decisions or usefu
 
 ## Slack
 
-The project's Slack channel is the **#lucene-dev** channel in the **the-asf** organization. Link: <https://the-asf.slack.com/messages/CE70MDPMF>
+- The project's Slack channel are the `#lucene-dev` and `#solr-dev` channels in the `the-asf` organization. These are primarily for developer discussions and not meant as support channels. Link: <https://the-asf.slack.com/messages/CE70MDPMF>
+- For Solr support, there is a (community maintained/unofficial) Slack organization that relays messages bi-directionally to/from the officially supported IRC channels. Link: https://s.apache.org/solr-slack
+- For relevance related discussions (Solr or other search engines), there's an unofficial Slack organization: https://opensourceconnections.com/slack
```
Comment (epugh) [inline content/pages/core/discussion.md:91]: This slack communities name is "Relevance Slack", the invite code just happens to be through OSC.   So how about "For relevance related discussions (Solr or other search engines), join Relevance Slack: http://opensourceconnections.com/slack"

**Comment by anshumg** [comment]
Thanks for adding this information, @chatman 

Can you please highlight and may be list the unofficial channels separately? You may list those as "Third Party" perhaps? 
You've mentioned this already but just wanted to be sure that the channels aren't perceived as official ones even accidentally.

**Comment by chatman** [comment]
Thanks for the review, @epugh , @anshumg !

**Review by anshumg — APPROVED** [review]

**Comment by janhoy** [comment]
Why was Solr slack info added to the Lucene-java webpage? https://lucene.apache.org/core/discussion.html#slack
That page does not list any other Solr mailing lists or resources.

And just curious - how did you merge this PR to production branch? When I compare production and master branches the edits show up as diffs.

Finally, please delete the now-unused feature branch.

**Comment by chatman** [comment]
> And just curious - how did you merge this PR to production branch? When I compare production and master branches the edits show up as diffs.

I merged the PR to master and cherry-picked the commits to production.

**Comment by chatman** [comment]
> Finally, please delete the now-unused feature branch.
Done.

**Comment by anshumg** [comment]
Cherry picking seems to cause issues with the workflow that is documented. The best way is to not cherry pick but at any point just create a PR from master to production, and nothing will get missed from being merged plus the histories would be same in terms of chronology of commits.

The ideal steps are:

1. Create a PR from your branch to apache/lucene-site:master
2. Merge the PR into master
3. Create a PR from master -> production

**Comment by chatman** [comment]
This is very cumbersome, esp for a RM during a release.

**Comment by chatman** [comment]
> Cherry picking seems to cause issues with the workflow that is documented.

This is an unfounded assertion. Assuming master and production were both at sync before any of my changes, two commits added to master's tip (via this PR) when cherry-picked (without conflicts) to tip of production should be no different from a merge to production.

**Comment by anshumg** [comment]
I would like to think so as well but have just seen otherwise. Again, I'm fine with any workflow as long as it's convenient for everyone - no reason to add extra steps :) 

I'm not sure if cherry-pick messes up something as w.r.t. the code, it's identical.

**Comment by janhoy** [comment]
I just demonstrated this with #44 - it is really simple, just create the PR `master->production` and then merge it with *merge commit*, not a squash! This is now documented in README.

**Pull request closed without merging**